### PR TITLE
[feature](table-valued-function) support `hdfs()` tvf

### PR DESCRIPTION
**Pull request by BePPPower**
# Proposed changes

Issue Number: close #13959

This pr does two things:
1. support `hdfs()` table valued function.
2. add regression test

## Problem summary

Just like S3() table valued function, we can use `hdfs()` tvf to read file in hdfs. Like this:
```
MySQL [(none)]> select * from hdfs("uri" = "hdfs://127.0.0.1:842/user/doris/csv_format_test/student.csv", "fs.defaultFS" = "hdfs://127.0.0.1:8424", "hadoop.username" = "xx", "format" = "csv");
+------+---------+------+
| c1   | c2      | c3   |
+------+---------+------+
| 1    | alice   | 18   |
| 2    | bob     | 20   |
| 3    | jack    | 24   |
| 4    | jackson | 19   |
| 5    | liming  | 18   |
| 6    | luffy   | 20   |
| 7    | zoro    | 22   |
| 8    | sanzi   | 26   |
| 9    | wusuopu | 21   |
| 10   | nami    | 18   |
+------+---------+------+

// Also support desc function hdfs()
MySQL [(none)]> desc function hdfs("uri" = "hdfs://127.0.0.1:8424/user/doris/csv_format_test/student_with_names.csv", "fs.defaultFS" = "hdfs://127.0.0.1:8424", "hadoop.username" = "xx", "format" = "csv_with_names");
+-------+------+------+-------+---------+-------+
| Field | Type | Null | Key   | Default | Extra |
+-------+------+------+-------+---------+-------+
| id    | TEXT | Yes  | false | NULL    | NONE  |
| name  | TEXT | Yes  | false | NULL    | NONE  |
| age   | TEXT | Yes  | false | NULL    | NONE  |
+-------+------+------+-------+---------+-------+
```

And `hdfs()` is also support `csv/csv_with_names/csv_with_names_and_types/json/parquet/orc` file format.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
You need to add all new csv file to `.licenserc.yaml`, to pass the license check

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/HdfsTableValuedFunction.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    public static String HADOOP_FS_NAME = "fs.defaultFS";
+    // simple or kerberos
+    public static String HADOOP_SECURITY_AUTHENTICATION = "hadoop.security.authentication";
+    public static String HADOOP_USER_NAME = "hadoop.username";
+    public static String HADOOP_KERBEROS_PRINCIPAL = "hadoop.kerberos.principal";
+    public static String HADOOP_KERBEROS_KEYTAB = "hadoop.kerberos.keytab";
+    public static String HADOOP_SHORT_CIRCUIT = "dfs.client.read.shortcircuit";
+    public static String HADOOP_SOCKET_PATH = "dfs.domain.socket.path";
+
+    private static final ImmutableSet<String> FILE_FORMAT_PROPERTIES = new ImmutableSet.Builder<String>()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/HdfsTableValuedFunction.java:54]: `FILE_FORMAT_PROPERTIES` should be a common property for both HDFS and S3 table?
So how about move it to the parent class?

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.04 seconds
 load time: 441 seconds
 storage size: 17183690818 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221111155335_clickbench_pr_44572.html

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**